Dockerfile

```diff
@@ -10,11 +10,11 @@ COPY go.mod go.sum /go/node-feature-discovery/
 
 WORKDIR /go/node-feature-discovery
 
-RUN go mod download
-
 # Do actual build
 COPY . /go/node-feature-discovery
 
+RUN go mod download
+
```

Comment:
Doing this after the COPY nullifies the whole point of pre-loading go modules (for caching). Can't you copy the go.{mod,sum} of the new module above, similarly to the main module?